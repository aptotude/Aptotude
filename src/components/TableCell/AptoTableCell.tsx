import React from 'react';
import { StandardTypes } from '../../utils/standardTypes';
import classNames from 'classnames';
import './aptoTableCell.scss';

const COMPONENT_PREFIX = 'AptoTableCell';

export interface AptoTableCellDisplayProps extends StandardTypes {
  type?: 'th' | 'td' | undefined;
  scope?: 'col' | 'row' | undefined;
  highlight?: 'danger' | 'primary' | 'warning' | 'success' | undefined;
}

export class AptoTableCell extends React.Component<AptoTableCellDisplayProps> {
  public render() {
    const { type, highlight, scope, children, ...rest } = this.props;

    const Component = type || 'td';
    const classes = classNames(
      COMPONENT_PREFIX,
      highlight && `${COMPONENT_PREFIX}--${highlight}`
    );

    return (
      <Component scope={scope || undefined} className={classes} {...rest}>
        {children}
      </Component>
    );
  }
}
