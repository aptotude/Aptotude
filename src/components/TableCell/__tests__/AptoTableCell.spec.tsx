import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoTableCell } from '../AptoTableCell';

afterEach(() => {
  cleanup();
});

describe('AptoTableCell', () => {
  it('Renders Component', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell>hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders type th', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell type="th">hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders type td', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell type="td">hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra classes', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell className="foo">hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders scope col', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell scope="col">hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders scope row', () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <AptoTableCell scope="row">hi</AptoTableCell>
          </tr>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('highlight', () => {
    it('shows danger', () => {
      const { container } = render(
        <table>
          <tbody>
            <tr>
              <AptoTableCell highlight="danger">hi</AptoTableCell>
            </tr>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows primary', () => {
      const { container } = render(
        <table>
          <tbody>
            <tr>
              <AptoTableCell highlight="primary">hi</AptoTableCell>
            </tr>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows warning', () => {
      const { container } = render(
        <table>
          <tbody>
            <tr>
              <AptoTableCell highlight="warning">hi</AptoTableCell>
            </tr>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows success', () => {
      const { container } = render(
        <table>
          <tbody>
            <tr>
              <AptoTableCell highlight="success">hi</AptoTableCell>
            </tr>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
