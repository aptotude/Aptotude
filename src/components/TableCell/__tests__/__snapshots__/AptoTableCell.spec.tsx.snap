// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTableCell Renders Component 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell Renders extra classes 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="foo"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell Renders scope col 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell"
        scope="col"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell Renders scope row 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell"
        scope="row"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell Renders type td 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell Renders type th 1`] = `
<table>
  <tbody>
    <tr>
      <th
        class="AptoTableCell"
      >
        hi
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell highlight shows danger 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell AptoTableCell--danger"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell highlight shows primary 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell AptoTableCell--primary"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell highlight shows success 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell AptoTableCell--success"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableCell highlight shows warning 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="AptoTableCell AptoTableCell--warning"
      >
        hi
      </td>
    </tr>
  </tbody>
</table>
`;
